internal/sql/pg/fns.go

```diff
@@ -0,0 +1,114 @@
+package pg
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/kwilteam/kwil-db/common/sql"
+)
+
+const (
+	notifChannel       = "kwild_internal_notif"
+	sqlCreateFuncError = `CREATE OR REPLACE FUNCTION error(msg text)
+		RETURNS void AS $$
+		BEGIN
+			RAISE EXCEPTION '%', msg;
+		END;
+		$$ LANGUAGE plpgsql;`
+
+	sqlCreateFuncNotice = `CREATE OR REPLACE FUNCTION notice(payload text)
+		RETURNS void AS $$
+		DECLARE txid bigint;
+		BEGIN
+			SELECT txid_current() INTO txid;
```

Comment:
I think an assumption made here and in ABCI's FinalizeBlick is that `txid_current` will be an ID for the nested transactions, but it only pertains to the outermost transaction.  So all blockchain transactions in a block will have the same `pgtx:<txid>` prefix.